From 5a331016fc44986a3662ee022e0af6a796eee6a0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Timur=20Krist=C3=B3f?= <timur.kristof@gmail.com>
Date: Tue, 9 Jul 2019 08:05:20 +0200
Subject: [PATCH 2/2] Add build option to disable regenerating the metadata.

---
 cpp/CMakeLists.txt | 33 ++++++++++++++++++++++++---------
 1 file changed, 24 insertions(+), 9 deletions(-)

diff --git a/cpp/CMakeLists.txt b/cpp/CMakeLists.txt
index 1b9bdb79..7bc597e6 100644
--- a/cpp/CMakeLists.txt
+++ b/cpp/CMakeLists.txt
@@ -87,6 +87,7 @@ option ("USE_RE2" "Use RE2" "OFF")
 option ("USE_STD_MAP" "Force the use of std::map" "OFF")
 option ("BUILD_STATIC_LIB" "Build static libraries" "ON")
 option ("USE_PROTOBUF_LITE" "Link to protobuf-lite" "OFF")
+option ("REGENERATE_METADATA" "Regenerate metadata instead of using it from the source tree" "ON")
 
 if (${USE_ALTERNATE_FORMATS} STREQUAL "ON")
   add_definitions ("-DI18N_PHONENUMBERS_USE_ALTERNATE_FORMATS")
@@ -137,8 +138,10 @@ endif ()
 find_required_program (PROTOC protoc
                        "Google Protocol Buffers compiler (protoc)")
 
-find_required_program (JAVA java
-                       "Java Runtime Environment")
+if (${REGENERATE_METADATA} STREQUAL "ON")
+  find_required_program (JAVA java
+                         "Java Runtime Environment")
+endif ()
 
 if (APPLE)
   FIND_LIBRARY (COREFOUNDATION_LIB CoreFoundation)
@@ -278,14 +281,26 @@ function (add_metadata_gen_target TARGET_NAME
   set (JAR_PATH "${CMAKE_SOURCE_DIR}/../tools/java/cpp-build/target")
   set (JAR_PATH "${JAR_PATH}/cpp-build-1.0-SNAPSHOT-jar-with-dependencies.jar")
 
-  add_custom_command (
-    COMMAND ${JAVA_BIN} -jar
-      ${JAR_PATH} BuildMetadataCppFromXml ${XML_FILE}
-      ${CMAKE_SOURCE_DIR}/src/phonenumbers ${METADATA_TYPE}
+  if (${REGENERATE_METADATA} STREQUAL "ON")
+    add_custom_command (
+      COMMAND ${JAVA_BIN} -jar
+        ${JAR_PATH} BuildMetadataCppFromXml ${XML_FILE}
+        ${CMAKE_SOURCE_DIR}/src/phonenumbers ${METADATA_TYPE}
 
-    OUTPUT ${GEN_OUTPUT}
-    DEPENDS ${XML_FILE}
-  )
+      OUTPUT ${GEN_OUTPUT}
+      DEPENDS ${XML_FILE}
+    )
+  else ()
+    add_custom_command (
+      COMMAND echo "skip metadata generation from"
+        ${XML_FILE} "to"
+        ${CMAKE_SOURCE_DIR}/src/phonenumbers ${METADATA_TYPE}
+
+      OUTPUT ${GEN_OUTPUT}
+      DEPENDS ${XML_FILE}
+    )
+  endif ()
+  
   add_custom_target (
     ${TARGET_NAME}
     DEPENDS ${GEN_OUTPUT}
-- 
2.21.0

